HIP/hip-415.md

```diff
@@ -0,0 +1,255 @@
+| hash | Services | Stored in services (last 256 blocks). It is the 32 byte prefix of the runningHash of the previous record file. That is the running hash of the last record stream object from the previous record file. It is exposed to the 1) EVM during Transaction Execution through the BLOCKHASH opcode; 2) In the record file as the End Object Running Hash |
+| baseFeePerGas | Mirror Node | Always zero, since there is no EIP-1559 style floating block capacity fees in Hedera. |
+  | difficulty | Mirror Node | Hardcoded by Mirror Node(s) to hex encoded 0 |
+  | extraData | Mirror Node | Hardcoded by Mirror Nodes(s) to 0x |
+  | gasLimit | Mirror Node |  Computed by Mirror Node(s). The gas throttle limit per second multiplied by the target block time. |
+  | gasUsed | Mirror Node | Computed by Mirror Node(s). The sum of the gasUsed value for all ContractCall and ContractCreate transactions within the block. |
+  | logsBloom | Mirror Node | Computed by Mirror Node(s). It is the bloom filter for the logs within the block. |
+  | miner | Mirror Node | Hardcoded by the Mirror Nodes to the 0.0.98 address in the long zero prefix format.  |
+  | mixHash | Mirror Node | Hardcoded by Mirror Node(s) to 0x |
+  | nonce | Mirror Node | Hardcoded by Mirror Node(s) to hex-encoded 0 |
+  | parentHash | Mirror Node | The hash of the previous block. |
+  | receiptsRoot | N/A | Hardcoded by Mirror Node(s) to hex-encoded 0. It is expected that in the future it will be 
+populated once traceability is implemented. |
+  | sha3Uncles | Mirror Node | Hardcoded by Mirror Node(s) to the SHA3 computation of empty array (`0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347`) |
+  | size | Mirror Node | Computed by Mirror Node(s). The size of the record file. |
+  | stateRoot | Mirror Node | Hardcoded by Mirror Node(s) to hex-encoded 0. May contain state proof data in the future |
+  | totalDifficulty | Mirror Node | Hardcoded by Mirror Nodes(s) to hex-encoded 0 |
+  | transactions | Mirror Node | Computed by Mirror Node(s) by ingesting the record stream and aggregating RecordStreamObjects of type ContractCall and ContractCreate within the block. |
+  | transactionsRoot | Mirror Node | The same value as block `hash` |
+  | uncles | Mirror Node | Hardcoded by Mirror Node(s) to empty array [] |
+
+## Backwards Compatibility
+
+The following breaking changes will be introduced with the implementation of the HIP:
+
+- `BLOCKHASH` will no longer return an empty `hash` but the actual `hash` of the block as per the EVM specification.
+- `NUMBER` will no longer return the `timestamp` of the `block`, but rather the proper `block` number as specified in the HIP.
```

Comment:
So this makes a commitment for us that we will never use the inter-block times for number, and that we will know when the block counter increments prior to the execution of the EVM.

Reply:
Yes, that is correct.
The block updates are occurring at the and of the `handleTransaction` thread post Transaction Execution, at the point where we export the TX Record for streaming. At that point, the check for the new block/record file is executed. If new block/file is to be created, the transaction is handled and exported will be included in the new block and will "close" the previous one. 

We must verify that the correct `timestamp`, `blockhash` and `number` are set and returned during the execution of the first transaction within the block. It might be the case that the execution is performed against the "old" values  (aka parent block data).